### PR TITLE
Add keygen autorun to tests

### LINKED ISSUE #30: cargo test --workspace fails in my local dev environment
I see this commit locally:   b13c2a22b0ab730308e90437ea53dd902029d6b1

### PR DESCRIPTION
**Pull request by AloeareV**
Fixes #30

### CODE REVIEW

File: lib/keygen.sh [hunk 1]
```diff
@@ -3,4 +3,5 @@ openssl req -x509 -out localhost.crt -keyout localhost.key \
     -subj '/CN=localhost' -extensions EXT -config <( \
        printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth,clientAuth")
 
-cat localhost.crt localhost.key > lib/localhost.pem
\ No newline at end of file
+cat localhost.crt localhost.key > localhost.pem
+rm localhost.crt localhost.key
```
Comment (zancas) [inline lib/keygen.sh:7]: Why remove these?  Let's leave them in, if they're going to provide a history in an emphemeral test-specific directory.

File: lib/src/lightclient/test_server.rs [hunk 2]
```diff
@@ -40,6 +42,16 @@ pub async fn create_test_server(
     oneshot::Sender<()>,
     JoinHandle<()>,
 ) {
+    KEYGEN.call_once(|| {
```
Comment (zancas) [inline lib/src/lightclient/test_server.rs:45]: Neato!

**Comment by zancas** [comment]
Let's move ahead with making per-test state unique, and easily found.

**Review by zancas — APPROVED** [review]

**Comment by AloeareV** [comment]
Wait, nevermind, my CI fix made it in before the merge.

**Comment by AloeareV** [comment]
This fixed #9

**Pull request merged**